freud/box.py

```diff
@@ -204,16 +206,11 @@ def make_absolute(self, fractional_coordinates, out=None):
         flatten = fractions.ndim == 1
         fractions = np.atleast_2d(fractions)
         fractions = freud.util._convert_array(fractions, shape=(None, 3))
-        out = freud.util._convert_array(
-            out, shape=fractions.shape, allow_copy=False)
+        out = freud.util._convert_array(out, shape=fractions.shape, allow_copy=False)
 
-        cdef const float[:, ::1] l_points = fractions
-        cdef unsigned int Np = l_points.shape[0]
-        cdef float[:, ::1] l_out = out
+        Np = fractions.shape[0]
 
-        self.thisptr.makeAbsolute(
-            <vec3[float]*> &l_points[0, 0], Np,
-            <vec3[float]*> &l_out[0, 0])
+        self._cpp_obj.makeAbsolute(fractions, Np, out)
```

Comment:
Continuing the discussion about `_convert_array` - there is no need to access `Np = fractions.shape[0]` here (and in similar methods). That check can be made in C++. In most cases, I would hope that the input "array" provided by the user can be passed directly in to the C++ method.

To preserve full `API` compatibility with freud 2.0 in this specific method, the `flatten` code above would need to be wrapped in some sort of type check.
```
        flatten = fractions.ndim == 1
        fractions = np.atleast_2d(fractions)
```
I don't know what specific API use-case the `atleast_2d` achieves. I see that it is used extensively in the box module. Is that true in the rest of freud?

Reply:
freud chooses to allow users to give and receive arrays containing only a single position as a 1d array instead of 2d if they want. Much of the pre/post processing via calls to `atleast_2d` (which adds a dimension to a 1d array so its 2d) and `squeeze` is to support this use case.

This happens throughout the box module, but there is only one other call to `atleast_2d` in freud, which is located in the `NeighborQuery.query` method.